lib/fast_jsonapi/object_serializer.rb

```diff
@@ -292,6 +294,38 @@ def create_relationship(base_key, relationship_type, options, block)
         )
       end
 
+      ##
+      # Add an extra layer for attribute filtering to the serializer operating on the full list of defined attributes,
+      # but before any fieldset is applied. This is a more performant option than providing conditionals to every attribute.
+      #
+      # @param filter_method_name [Symbol, nil]
+      #     The name of a class method used to filter the set of attributes. This method will receive the superset of attributes,
+      #     the current record getting serialized and the serializer parameters passed along.
+      #
+      # @yieldparam filter_block [#call]
+      #     If a block is provided instead of a method name, this is going to be called when building the attributes hash.
+      #     The arguments to the block are the same as for the method: the superset of attributes, the record getting serialized
+      #     and the serializer parameters.
+      def attributes_filter(filter_method_name = nil, &block)
+        self.attributes_filter_method = filter_method_name || block
```

Comment:
You should raise an argument error if both a method name and a block are provided